numba/typing/templates.py

```diff
@@ -604,6 +611,17 @@ def _resolve(self, typ, attr):
         return sig.return_type
 
 
+def _adjust_omitted_args(argtypes, argvals):
+    """Add dummy arguments for each missing types.Omitted in *argtypes*.
+    """
+    if len(argtypes) > len(argvals):
```

Comment:
Is this branch ever executed?

Reply:
Yes, kwargs support requires this branch to patch the arguments.  When the signature contains Omitted types, those locations are missing in the actual arguments.  This re-adds them as required by anything base on `Dispatcher`.

Reply:
you're right.  It turns out adjusting the pysignature fixes the actual problem already.